Adds adaptive allocations feature description to conceptual docs

szabosteve opened the pull request:
## Overview

This PR describes adaptive allocations in the `ml-nlp-shared.asciidoc` file. The feature description is then single-sourced from there in the ELSER, E5, and deploy trained model conceptual docs.

### Preview

* [ELSER](https://stack-docs_bk_2763.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-nlp-elser.html#elser-adaptive-allocations)
* [E5](https://stack-docs_bk_2763.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-nlp-e5.html#e5-adaptive-allocations)
* [Deploy trained models](https://stack-docs_bk_2763.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-nlp-deploy-model.html#nlp-model-adaptive-allocations)

szabosteve commented on docs/en/stack/ml/nlp/ml-nlp-shared.asciidoc:
```diff
@@ -1,3 +1,22 @@
+You can enable adaptive allocations by using:
+
+* the Create inference endpoint API for {ref}/infer-service-elser.html[ELSER], {ref}/infer-service-elasticsearch.html[E5 and models uploaded through Eland] that are used as {infer} services.
+* the {ref}/start-trained-model-deployment.html[start trained model deployment] or {ref}/update-trained-model-deployment.html[update trained model deployment] APIs for trained models that are uploaded to and managed on {ml} nodes.
```
> for trained models that are uploaded to and managed on {ml} nodes.

@jan-elastic I need some input for phrasing this more accurately.

jan-elastic replied:
I would replace "are uploaded to and managed on" by "are deployed on".

jan-elastic commented on docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc:
```diff
@@ -186,7 +186,9 @@ NOTE: Since eland uses APIs to deploy the models, you cannot see the models in
-number of threads used by each allocation of your deployment.
+number of threads used by each allocation of your deployment. Alternatively, you
+can enable <<nlp-model-adaptive-allocations>> to automatically create and remove
+model allocations based on the current workload of the model.
```
To be precise: adaptive allocations is an alternative to setting the number of allocations. It's not an alternative to setting the number of threads. That has to be done regardless.

jan-elastic commented on docs/en/stack/ml/nlp/ml-nlp-shared.asciidoc:
```diff
@@ -1,3 +1,22 @@
+tag::ml-nlp-adaptive-allocations[]
+The numbers of threads and allocations you can set manually for a model remain constant even when not all the available resources are fully used or when the load on the model requires more resources.
+Adaptive allocations, however, provide full control and flexibility over the model allocations, and help you to manage performance and cost more easily.
```
Not so sure about this "full control and flexibility": you actually give away control of the number of allocations.

What about: "Instead of setting the number of allocations manually, you can enable adaptive allocations to set the number of allocations based on the load on the process. This can help you to manage performance and cost more easily."

jan-elastic reviewed (commented):
couple of small things. in general lgtm

szabosteve commented:
@jan-elastic Thanks for your review, I addressed your comments via [1de8986](https://github.com/elastic/stack-docs/pull/2763/commits/1de89862e7cc6cb8a0de96bb6a426fde4e0348d9).

jan-elastic reviewed (approved):
LGTM
